x-pack/auditbeat/module/system/socket/socket_linux.go

```diff
@@ -86,20 +88,38 @@ func init() {
 	}
 }
 
+var currentInstance *MetricSet
+
 // New constructs a new MetricSet.
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
-	cfgwarn.Beta("The %s dataset is beta.", fullName)
-
 	config := defaultConfig
 	if err := base.Module().UnpackConfig(&config); err != nil {
 		return nil, errors.Wrapf(err, "failed to unpack the %s config", fullName)
 	}
+	if currentInstance != nil {
```

Comment:
does this need to be wrapped in a mutex or is `New` always called from a single goroutine?

Reply:
It's called from a single goroutine, but in retrospect it makes more sense to add a mutex, just in case